feat: introducing "copy_terraform_lock_file" to fine tune Lock File Handling (#2889)
gruntwork-io/terragrunt: docs/_docs/02_features/lock-file-handling.md

Also, any time you change the providers you're using, and re-run `init`, the lock file will be updated, so make sure
to check the updates into version control too.

### Disabling the copy of the generated lock file

In certain use cases, like when using a remote module containing a lock file within it, you probably

> don't want Terragrunt to also copy the lock file into your working directory. In these scenarios you can opt-out of copying

Reviewer: ```suggestion
don't want Terragrunt to also copy the lock file into your working directory. In these scenarios, you can opt-out of copying
```

> don't want Terragrunt to also copy the lock file into you working directory, in such scenarios you can opt out from the copy

Reviewer 2: ```suggestion
don't want Terragrunt to also copy the lock file into your working directory. In these scenarios you can opt-out of copying
```

> feature by using `copy_terraform_lock_file = false` in the `terragrunt.hcl` file as following:

Reviewer 2: ```suggestion
the `.terraform.lock.hcl` file by using `copy_terraform_lock_file = false` in the `terraform` configuration block as follows:
```
